### PR TITLE
feat: add `Copy-vSphereRole`

### LINKED ISSUE #512: VCF452: Issue with assigning privileges to HCX vSphere role using `Add-vSphereRole`
### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware/power-validated-solutions-for-cloud-foundation/blob/main/CODE_OF_CONDUCT.md).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### VMware Cloud Foundation

4.5.2

### Module Version

2.9.0.1016

### PowerShell Version

7.4.1

### PowerCLI Version

13.2.1.22851661

### PowerVCF Version

2.4.1.1000

### Guest Operating System

windows server 2019 Datacenter

### Environment Details

_No response_

### Description

Tried to perform the PowerShell procedure mentioned in the **[Define a Custom Role in vSphere for Cloud-Based Workload Protection for VMware Cloud Foundation](https://docs.vmware.com/en/VMware-Cloud-Foundation/services/vcf-cloud-based-workload-protection-v1/GUID-0F8534BE-1E4F-4C56-956E-E59693AF9B7C.html#GUID-0F8534BE-1E4F-4C56-956E-E59693AF9B7C)** section. But it failed with an error message as shown.

### Error or Debug Output

Console shows the following error message while running the cmd.

```powershell
Add-vSphereRole -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -sddcDomain $sddcDomainName -roleName $hcxRole
Get-VIPrivilege: 2/20/2024 11:55:58 Get-VIPrivilege  VIPrivilege with id 'Host.Sgx.Register' was not found using the specified
filter(s).
 Error at Script Line 25578
 Relevant Command: Set-VIRole -Server $vcfVcenterDetails.fqdn -Role $roleName -AddPrivilege (Get-VIPrivilege -ID $privilege) -Confirm:$False -ErrorAction SilentlyContinue | Out-Null
 Error Message: Cannot validate argument on parameter 'AddPrivilege'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.
 ```

### Expected Behavior

Execute without a failure

### Actual Behavior

The same error appears for some more privileges. 

It seems like the following privileges are not applicable for vCenter Server 7.x and should be removed from `hcx-vsphere-integration-7x.role` for `Add-vSphereRole` cmdlet to work on VCF452 (vCenter Server 7.x). 

```console
Host.Sgx.Register
Host.Stats.Query
Host.Tpm.Read
Host.Tpm.Unseal
PartnerRestDaemon.Read
PartnerRestDaemon.Write
Sessions.CollectPrivilegeChecks
StorageProfile.Apply
StorageProfile.EditPermissions
StorageProfile.ViewPermissions
VirtualMachine.DataSets.DataSetCreate
VirtualMachine.DataSets.DataSetDelete
VirtualMachine.DataSets.DataSetEntryDelete
VirtualMachine.DataSets.DataSetEntryGet
VirtualMachine.DataSets.DataSetEntryList
VirtualMachine.DataSets.DataSetEntrySet
VirtualMachine.DataSets.DataSetGet
VirtualMachine.DataSets.DataSetList
VirtualMachine.DataSets.DataSetUpdate
vSANStats.Access
```

### Steps to Reproduce

Try to run the following cmdlet and provide hcx-vsphere-integration-7x.role as an input file.

```powershell
$sddcManagerFqdn = "sfo-vcf01.sfo.rainpole.io"
$sddcManagerUser = "administrator@vsphere.local"
$sddcManagerPass = "VMw@re1!"
$sddcDomainName = "sfo-w01"
$hcxRole = "VMware HCX to vSphere Integration"

Add-vSphereRole -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -sddcDomain $sddcDomainName -roleName $hcxRole
```

### Log Fragments and Files

_No response_

### Screenshots

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by tenthirtyam**
### Summary

- Adds `Copy-vSphereRole` to request the set of privileges assigned to a specific role in vCenter Server and create a new role with the same privileges. This will be used to replace use of `.role` files where the list of privileges can change release over release for an system managed role that is the basis for a new role.
- Removes role files VMware Aria Automation Orchestrator and VMware HCX. Use the `Copy-vSphereRole` cmdlet.

### Type

- [ ] Bugfix
- [x] Enhancement or Feature
- [ ] Code Style or Formatting
- [ ] Documentation
- [ ] Refactoring
- [ ] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### Test and Documentation

- [x] Tests have been completed.
- [x] Documentation has been added or updated.

### Issue References

Closes #512

### CODE REVIEW

File: docs/documentation/functions/vsphere/Copy-vSphereRole.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+```
+
+### -ProgressAction
+
+Progress Action
+
+```yaml
+Type: ActionPreference
+Parameter Sets: (All)
+Aliases: proga
```
Comment (andybeltz) [inline docs/documentation/functions/vsphere/Copy-vSphereRole.md:138]: Is this Alias supposed to be proga or progress?

File: docs/documentation/functions/vsphere/Copy-vSphereRole.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+Required: False
+Position: Named
```
Comment (andybeltz) [inline docs/documentation/functions/vsphere/Copy-vSphereRole.md:141]: Should this be Position: 7?

**Review by andybeltz — CHANGES_REQUESTED** [review]
Check out the two comments I made, to ensure that the changes are accurately portrayed in the code.

File: docs/documentation/functions/vsphere/Copy-vSphereRole.md (hunk 1 above)
Comment (tenthirtyam) [reply docs/documentation/functions/vsphere/Copy-vSphereRole.md:138]: Thanks! This is the output from the PlatyPS module that generates the documentation from the function.

File: docs/documentation/functions/vsphere/Copy-vSphereRole.md (hunk 2 above)
Comment (tenthirtyam) [reply docs/documentation/functions/vsphere/Copy-vSphereRole.md:141]: Thanks! This is the output from the PlatyPS module that generates the documentation from the function.

**Pull request merged**